### PR TITLE
Includes

### PR DESCRIPTION
**Pull request by PhyberApex**
Fix for #198 
Also did a cleanup of the LINQ stuff

### CODE REVIEW

File: Saule/Serialization/ResourceSerializer.cs [hunk 1]
```diff
@@ -192,22 +190,13 @@ private JToken SerializeData(ResourceGraph graph)
...
                     return new JArray();
                 }
-                else
-                {
-                    return JArray.FromObject(tokens);
-                }
+                return JArray.FromObject(tokens);
             }
-            else
+            if (!tokens.Any())
```
Comment (joukevandermaas) [inline Saule/Serialization/ResourceSerializer.cs]: I think this part is the real bug fix?

File: Saule/Serialization/ResourceSerializer.cs [hunk 2]
```diff
@@ -377,43 +363,35 @@ private JObject SerializeRelationships(ResourceGraphNode node)
 
         private JToken SerializeRelationshipData(ResourceGraphNode node, ResourceGraphRelationship relationship)
         {
-            // short circuit if not included in graph
-            if (!relationship.Included)
```
Comment (joukevandermaas) [inline Saule/Serialization/ResourceSerializer.cs]: Are you sure this part can go?

**Comment by PhyberApex** [comment]
Will write some tests as soon as I get some free time for it and will check if it is safe to remove that part.

~Cheers

**Comment by PhyberApex** [comment]
Hey I updated this branch as we need it internally as well...hope that's okay. I also wrote the test to check this works correctly. Hope that's all good.

~Cheers

**Pull request merged**